File: backend/src/main/java/ma/yassine/digitalbanking/services/ChatbotService.java
package ma.yassine.digitalbanking.services;

import ma.yassine.digitalbanking.dtos.ChatRequest;
import ma.yassine.digitalbanking.dtos.ChatResponse;

public interface ChatbotService {

    ChatResponse chat(ChatRequest request);
    void ingestDocumentation(String documentation);
}

File: frontend/src/app/core/services/chatbot.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { ChatRequest, ChatResponse } from '../models/chat.model';

@Injectable({ providedIn: 'root' })
export class ChatbotService {
  private url = `${environment.apiUrl}/chat`;

  constructor(private http: HttpClient) {}

  chat(request: ChatRequest): Observable<ChatResponse> {
    return this.http.post<ChatResponse>(this.url, request);
  }
}
